src/kinesis.js

```diff
@@ -1,6 +1,224 @@
 const AWS = require('aws-sdk');
+const uuid = require('uuid');
+const Docker = require('dockerode');
+const cloneDeep = require('lodash/cloneDeep');
+const fromPairs = require('lodash/fromPairs');
+const NestedError = require('nested-error-stacks');
+const promiseRetry = require('promise-retry');
 
-exports.useKinesis = (test, streamName) => {
+const { getHostAddress, ensureImage } = require('./docker');
+const Environment = require('./Environment');
+
+const KINESIS_IMAGE = 'localstack/localstack:latest';
+
+let kinesisStreams = [];
+
+function streams (streams) {
+  kinesisStreams = streams.map(StreamName => ({
+    StreamName,
+    ShardCount: 1
+  }));
+}
+
+function getStreamName (streamName, uniqueIdentifier) {
+  return uniqueIdentifier ? `${streamName}-${uniqueIdentifier}` : streamName;
+}
+
+async function createStreams (kinesisClient, uniqueIdentifier) {
+  const failedProvisons = [];
+  await Promise.all(
+    kinesisStreams.map(async stream => {
+      const newStream = cloneDeep(stream);
+      const StreamName = getStreamName(newStream.StreamName, uniqueIdentifier);
+      newStream.StreamName = StreamName;
+
+      try {
+        await kinesisClient.createStream(newStream).promise();
+        await kinesisClient.waitFor('streamExists', { StreamName }).promise();
+      } catch (err) {
+        failedProvisons.push(StreamName);
+        console.error(`Failed to create stream "${StreamName}"`, err);
+      }
+    })
+  );
+
+  if (failedProvisons.length) {
+    throw new Error(`Failed to create streams: ${failedProvisons.join(', ')}`);
+  }
+}
+
+async function destroyStreams (kinesisClient, uniqueIdentifier) {
+  const failedDeletions = [];
+  const { StreamNames } = await kinesisClient.listStreams().promise();
+  const schemaTableNames = kinesisStreams
+    .map(({ StreamName }) => getStreamName(StreamName, uniqueIdentifier));
+  const tablesToDestroy = StreamNames
+    .filter(name => schemaTableNames.includes(getStreamName(name, uniqueIdentifier)));
+
+  await Promise.all(
+    tablesToDestroy
+      .map(async StreamName => {
+        try {
+          await kinesisClient.deleteStream({ StreamName }).promise();
+          await kinesisClient.waitFor('streamNotExists', { StreamName }).promise();
+        } catch (err) {
+          failedDeletions.push(StreamName);
+          console.error(`Failed to destroy stream "${StreamName}"`, err);
+        }
+      })
+  );
+
+  if (failedDeletions.length) {
+    throw new Error(`Failed to destroy streams: ${failedDeletions.join(', ')}`);
+  }
+}
+
+function buildStreamNameMapping (uniqueIdentifier) {
+  return fromPairs(kinesisStreams.map(({StreamName}) => {
```

Comment:
Just FYI, you could return a proper `Map` with `new Map(<array of pairs>)`.

Reply:
I'm just following DynamoDB in this case.  I can change it if that is preferable.

Reply:
Is it okay if I leave this as is?

Reply:
Ya, it's not a big deal, was just pointing it out.